socketio/src/payload.rs

```diff
@@ -1,31 +1,45 @@
 use bytes::Bytes;
+use serde_json::Value;
 
 /// A type which represents a `payload` in the `socket.io` context.
-/// A payload could either be of the type `Payload::Binary`, which holds
-/// data in the [`Bytes`] type that represents the payload or of the type
-/// `Payload::String` which holds a [`std::string::String`]. The enum is
-/// used for both representing data that's send and data that's received.
+/// The enum is used for both representing data that's send and
+/// data that's received.
 #[derive(Debug, PartialEq, Eq, Clone)]
+#[non_exhaustive]
 pub enum Payload {
     Binary(Bytes),
-    String(String),
+    Json(Value),
```

Comment:
nit: consider adding a way to have some sort of way to convert a struct that impls Deserialize into Payload.